plugins/woocommerce-admin/client/guided-tours/add-product-tour/index.tsx

```diff
@@ -287,8 +287,8 @@ export const ProductTour = () => {
 				} );
 			}
 		},
-		onNextStepHandler: ( stepIndex ) => {
-			const stepName = tourConfig.steps[ stepIndex ].meta.name;
+		onNextStepHandler: ( newStepIndex ) => {
+			const stepName = tourConfig.steps[ newStepIndex - 1 ].meta.name;
```

Comment:
Before `@automattic/tour-kit@1.1.2`, the previous index was sent instead of the actual new index. After this update, the correct index is sent to the `onGoToStep`, `onPreviousStep`, and `onNextStep` callbacks.